mm2src/coins/utxo/utxo_balance_events.rs

```diff
@@ -0,0 +1,144 @@
+        const RECEIVER_DROPPED_MSG: &str = "Receiver is dropped, which should never happen.";
+
+        let ctx = match MmArc::from_weak(&self.as_ref().ctx) {
+            Some(ctx) => ctx,
+            None => {
+                let msg = "MM context must have been initialized already.";
+                tx.send(EventInitStatus::Failed(msg.to_owned()))
+                    .expect(RECEIVER_DROPPED_MSG);
+                panic!("{}", msg);
+            },
+        };
+
+        let addresses = match self.my_addresses().await {
+            Ok(t) => t,
+            Err(e) => {
+                tx.send(EventInitStatus::Failed(e.to_string()))
+                    .expect(RECEIVER_DROPPED_MSG);
+                panic!("{}", e);
+            },
+        };
+
+        for address in addresses {
+            let script = output_script(&address, keys::Type::P2PKH);
+            let script_hash = electrum_script_hash(&script);
+            let scripthash = hex::encode(script_hash);
+
+            if let Err(e) = self
+                .as_ref()
+                .rpc_client
+                .blockchain_scripthash_subscribe(scripthash)
+                .compat()
+                .await
+            {
+                tx.send(EventInitStatus::Failed(e.to_string()))
+                    .expect(RECEIVER_DROPPED_MSG);
+                panic!("{}", e);
+            }
+        }
+
+        tx.send(EventInitStatus::Success).expect(RECEIVER_DROPPED_MSG);
+
+        let mut current_balances: HashMap<String, BigDecimal> = HashMap::new();
+        loop {
+            match self
+                .as_ref()
+                .scripthash_notification_receiver
+                .as_ref()
+                .expect("Can not be `None`")
+                .lock()
+                .await
+                .try_next()
+            {
+                Ok(Some(_)) => {},
+                _ => {
+                    Timer::sleep(0.1).await;
+                    continue;
+                },
+            };
+
+            let new_balances = match self.my_addresses_balances().await {
+                Ok(t) => t,
+                _ => continue,
+            };
+
+            if new_balances == current_balances {
+                continue;
+            }
+
+            // Get the differences
+            let updated_parts: HashMap<String, BigDecimal> = new_balances
+                .iter()
+                .filter_map(|(key, new_value)| match current_balances.get(key) {
+                    Some(current_value) if new_value != current_value => Some((key.clone(), new_value.clone())),
+                    None => Some((key.clone(), new_value.clone())),
+                    _ => None,
+                })
+                .collect();
+
+            let balance_updates: Vec<_> = updated_parts
+                .iter()
+                .map(|(address, balance)| {
+                    json!({
+                        "ticker": self.ticker(),
+                        "address": address,
```

Comment:
I am not certain about whether this field is necessary or not. I did this because of the way of `HDWalletBalance` handles balances.

Reply:
It's necessary, it's even better if we used `coin_balance_report` https://github.com/KomodoPlatform/komodo-defi-framework/blob/173f8e323d81bd3946565d93b4ca32539f86e15f/mm2src/coins/coin_balance.rs#L117 but I am not sure how these events will be used yet in GUIs since HD wallet is not added there yet. So let's keep it as is for now, I just noticed that it would be good when requesting addresses balances to have some kind of filter or a different function so that we can request only the balance of the address that we get notified about to reduce request overhead since HD wallets can have a lot of addresses enabled.

Reply:
I added this comment to HD wallet checklist to not forget about it :) https://github.com/KomodoPlatform/komodo-defi-framework/issues/1838#issuecomment-1675467570

Reply:
P.S. I should modify tendermint balance events when HD wallet implementation is done for it too.

Reply:
> I just noticed that it would be good when requesting addresses balances to have some kind of filter or a different function so that we can request only the balance of the address that we get notified about to reduce requests cost/overhead since HD wallets can have a lot of addresses enabled.

If I understand you correctly, you sharing the same suggestion with Artem which I fixed as mentioned here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2013#discussion_r1409050394